Php develop

OmarAk312 opened the pull request:
- DB
- PHP Skript load data

OmarAk312 committed:
Version 1.0
Datenbank erstellen

OmarAk312 committed:
Verbindung mit der Datenbank herstellen.
Die Zugriff erfolgt dann über die Variable $con.

OmarAk312 committed:
angepasst

OmarAk312 committed:
Informationen aus der DB abrufen

janbellenberg commented on db/db.sql:
```diff
@@ -0,0 +1,36 @@
+    Ra_Nr INT,
+    Pos_Nr INT,
+    Art_ID INT,
+    FOREIGN KEY (Ge_Nr) REFERENCES Gebaeude(Ge_Nr),
```
on update / on delete fehlt

janbellenberg commented on db/db.sql:
```diff
@@ -0,0 +1,36 @@
+CREATE DATABASE messstation;
```
CREATE DATABASE IF NOT EXISTS `messstation` DEFAULT CHARACTER SET utf8 COLLATE utf8_general_ci;
USE messstation

janbellenberg commented on db/db.sql:
```diff
@@ -0,0 +1,36 @@
+
+CREATE TABLE Messung{
+    Messung_ID INT PRIMARY KEY AUTO_INCREMENT,
+    Zeitpunkt SMALLDATETIME,
```
SMALLDATETIME? -> DATE TIME

janbellenberg replied:
DEFAULT: current timestamp

janbellenberg commented on db/Connection.php:
```diff
@@ -0,0 +1,63 @@
+<?php
+
+class connection{
```
Connection groß geschrieben

janbellenberg commented on db/Data.php:
```diff
@@ -0,0 +1,58 @@
+                    $co2 = $row["Messung_Wert"];
+                    $co2_timestamp = $row["timestamp"];
+                }
+    
```
Ausgabe als JSON fehlt

janbellenberg reviewed (changes requested):
Muss noch überarbeitet werden

OmarAk312 committed:
Class-Name corrected

OmarAk312 committed:
return the array as JSON
to Avoid echo of empty string -->
Set HTTP response status code to: 500 - Internal Server Error

OmarAk312 committed:
corrected

OmarAk312 committed:
added default

OmarAk312 committed:
...

janbellenberg commented on db/Data.php:
```diff
@@ -29,30 +25,49 @@
                     $temperature_timestamp = $row["timestamp"];
+
+                    
+                    $response[0] = array(
```
Bitte die Response nicht als Array, sondern als Objekt mit 3 Attributen für die drei Arrays

janbellenberg commented on db/db.sql:
```diff
@@ -1,36 +1,38 @@
     Gebaeude_Name varchar(100) NOT NULL UNIQUE
-}
-
+)
```
Semikolon

janbellenberg commented on db/db.sql:
```diff
@@ -1,36 +1,38 @@
-CREATE TABLE Messung{
+CREATE TABLE Messung(
     Messung_ID INT PRIMARY KEY AUTO_INCREMENT,
     Zeitpunkt DATETIME,
```
Default Wert für neue Daten: current_timestamp()

janbellenberg commented on db/db.sql:
```diff
@@ -1,36 +1,38 @@
+    FOREIGN KEY (Ra_Nr) REFERENCES Raum(Ra_Nr) ON DELETE ON UPDATE,
+    FOREIGN KEY (Pos_Nr) REFERENCES Position(Pos_Nr) ON DELETE ON UPDATE,
+    FOREIGN KEY (Art_ID) REFERENCES Arten_Messungen(Art_ID) ON DELETE ON UPDATE
+)
```
Bitte noch einen extra Benutzer anlegen für die Verbindung

janbellenberg commented on db/Data.php:
```diff
@@ -2,15 +2,11 @@
     if (isset($_GET["data"])) {
         $data = $_GET["data"];
```
Das Feld heißt "date"

janbellenberg reviewed (changes requested):
Noch nicht perfekt

OmarAk312 committed:
tested DB

OmarAk312 committed:
ER-Modell zu "messstation" DB

OmarAk312 committed:
CURRENT_TIMESTAMP() modifed

OmarAk312 committed:
Version 2.0
DB Erstellung

OmarAk312 committed:
changed name

OmarAk312 committed:
Tabel "Arten_Messung" deleted

OmarAk312 committed:
input: date=2021-06-17&position=1
output: {"co2":"123","temperature":"1234","humidity":"32","timestamp":"14:53"}

OmarAk312 committed:
small adjustments

OmarAk312 committed:
examples

input: building=1
output: {"building":"HNN"}

input: room=1
output: {"room":"205"}

OmarAk312 committed:
New ER-Model

OmarAk312 committed:
Umbennung

OmarAk312 committed:
erwartet wird:
GET von der URL
Pos_Nr=...,
co2_Wert=...,
temperatur_Wert=...,
feuchtigkeit_Wert=...,

Ergebnis:
Die Daten werden in der Datenbank eingetragen.
Das Datum wird dort automatisch abgerufen und gespeichert.

Die Position sollte vorher schon festgelegt sein.

OmarAk312 committed:
...

OmarAk312 committed:
Um alle Positionen zu bekommen sollte man
positions in der URL setzen.

Um alle Räume zu bekommen sollte man
rooms in der URL setzen.

Um alle Gebäuden zu bekommen sollte man
locations in der URL setzen

OmarAk312 committed:
getestet alles in Ordnung

janbellenberg reviewed (approved):
Ist nun kompatibel mit der React Anwendung
